build/reactable.js

```diff
@@ -1239,12 +1271,21 @@ window.ReactDOM["default"] = window.ReactDOM;
                 for (var i = 0; i < children.length; i++) {
                     var data = children[i].props.data;
 
-                    for (var j = 0; j < this.props.filterable.length; j++) {
-                        var filterColumn = this.props.filterable[j];
-
-                        if (typeof data[filterColumn] !== 'undefined' && (0, _libExtract_data_from.extractDataFrom)(data, filterColumn).toString().toLowerCase().indexOf(filter) > -1) {
-                            matchedChildren.push(children[i]);
-                            break;
+                    for (var filterColumn in this._filterable) {
+                        if (typeof data[filterColumn] !== 'undefined') {
+                            // Default filter
+                            if (typeof this._filterable[filterColumn] === 'undefined' || this._filterable[filterColumn] === 'default') {
+                                if ((0, _libExtract_data_from.extractDataFrom)(data, filterColumn).toString().toLowerCase().indexOf(filter) > -1) {
+                                    matchedChildren.push(children[i]);
+                                    break;
+                                }
+                            } else {
+                                // Apply custom filter
+                                if (this._filterable[filterColumn]((0, _libExtract_data_from.extractDataFrom)(data, filterColumn).toString())) {
+                                    matchedChildren.push(children[i]);
+                                    break;
+                                }
+                            }
```

Comment:
these nested conditionals are maybe a bit messy so definitely open to input on how to improve or organize them